test/test_integrators.jl

```diff
@@ -68,64 +68,84 @@ u0 = [0.7611944793397108
-f1dofs = [1,2,3]
-f2dofs = [1,3]
-f3dofs = [1,3]
-fsplit2_inner = GenericSplitFunction((f3,f3), (f3dofs, f3dofs))
-fsplit2_outer = GenericSplitFunction((f1,fsplit2_inner), (f1dofs, f2dofs))
-
-timestepper_inner = LieTrotterGodunov(
-    (DummyForwardEuler(), DummyForwardEuler())
-)
-timestepper2 = LieTrotterGodunov(
-    (DummyForwardEuler(), timestepper_inner)
-)
-prob2 = OperatorSplittingProblem(fsplit2_outer, u0, tspan)
-integrator2 = DiffEqBase.init(prob2, timestepper2, dt=0.01, verbose=true)
-DiffEqBase.solve!(integrator2)
-
-DiffEqBase.reinit!(integrator2, u0; tspan)
-for (u, t) in DiffEqBase.TimeChoiceIterator(integrator2, 0.0:5.0:100.0)
+@testset "OperatorSplitting" begin
+    
+
+    # Note that we define the dof indices w.r.t the parent function.
+    # Hence the indices for `fsplit2_inner` are.
+    f1dofs = [1,2,3]
+    f2dofs = [1,3]
+    f3dofs = [1,3]
+    fsplit2_inner = GenericSplitFunction((f3,f3), (f3dofs, f3dofs))
+    fsplit2_outer = GenericSplitFunction((f1,fsplit2_inner), (f1dofs, f2dofs))
+    for TimeStepperType in (LieTrotterGodunov,)
+        for controller in (Thunderbolt.ReactionTangentController(0.5, 1.0, (0.01, 0.3)),) 
+            timestepper = TimeStepperType(
+                (DummyForwardEuler(), DummyForwardEuler())
+            )
+            timestepper_adaptive = Thunderbolt.AdaptiveOperatorSplittingAlgorithm(timestepper, controller)
+            timestepper_inner = TimeStepperType(
+                (DummyForwardEuler(), DummyForwardEuler())
+            )
+            timestepper_inner_adaptive = Thunderbolt.AdaptiveOperatorSplittingAlgorithm(timestepper_inner, controller) #TODO: Copy the controller instead
+            timestepper2 = TimeStepperType(
+                (DummyForwardEuler(), timestepper_inner)
+            )
+            timestepper2_adaptive = Thunderbolt.AdaptiveOperatorSplittingAlgorithm(timestepper2, controller)
+
+            for (tstepper1, tstepper_inner, tstepper2) in (
+                    (timestepper, timestepper_inner, timestepper2),
+                    # (timestepper_adaptive, timestepper_inner_adaptive, timestepper2_adaptive)
+                    )
+                # The remaining code works as usual.
+                integrator = DiffEqBase.init(prob, tstepper1, dt=0.01, verbose=true)
+                DiffEqBase.solve!(integrator)
+                ufinal = copy(integrator.u)
+                @test ufinal ≉ u0 # Make sure the solve did something
+
+                DiffEqBase.reinit!(integrator, u0; tspan)
+                for (u, t) in DiffEqBase.TimeChoiceIterator(integrator, 0.0:5.0:100.0)
+                end
+                @test  isapprox(ufinal, integrator.u, atol=1e-8)
+
+                DiffEqBase.reinit!(integrator, u0; tspan)
+                for (uprev, tprev, u, t) in DiffEqBase.intervals(integrator)
+                end
+                @test  isapprox(ufinal, integrator.u, atol=1e-8)
+
+                DiffEqBase.reinit!(integrator, u0; tspan)
```

Comment:
Maybe we should check if the retcode is not Success before solving, i.e.
```suggestion
                DiffEqBase.reinit!(integrator, u0; tspan)
                @test integrator.sol.retcode == DiffEqBase.ReturnCode.Default
```